validator/cycle/process_tasks.py

```diff
@@ -23,18 +23,32 @@
 from validator.utils.logging import TaskContext
 from validator.utils.logging import create_extra_log
 from validator.utils.logging import logger
+from validator.utils.minio import async_minio_client
 
 
-def _get_total_dataset_size(repo_name: str) -> int:
-    return int(sum(info.dataset_size for info in get_dataset_infos(repo_name).values() if info.dataset_size))
+async def _get_total_dataset_size(repo_name: str, file_format: FileFormat) -> int:
+    if file_format == FileFormat.S3:
+        bucket_name, object_name = async_minio_client.parse_s3_url(repo_name)
+        stats = await async_minio_client.get_stats(bucket_name, object_name)
+        size = stats.size
+    else:
+        size = sum(
+            info.dataset_size
+            for info in get_dataset_infos(repo_name).values()
```

Comment:
this one also calls an api so it would be good to await it with await to_thread() or similar

Reply:
Thanks I fixed and tested